Add section on pure SoA particles to documentation

atmyers opened the pull request:
The proposed changes:
- [ ] fix a bug or incorrect behavior in AMReX
- [ ] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [x] include documentation in the code and/or rst files, if appropriate

asalmgren committed:
use atomic add in SRD algorithm (#4754)

## Summary

## Additional background

## Checklist

The proposed changes:
- [X] fix a bug or incorrect behavior in AMReX
- [ ] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX
users
- [ ] include documentation in the code and/or rst files, if appropriate

atmyers committed:
Add section on pure SoA particles to documentation

atmyers committed:
fix spelling

ax3l commented on Docs/sphinx_documentation/source/Particle.rst:
```diff
@@ -135,24 +135,13 @@ See the figure :ref:`below<fig:particles:particle_arrays>` for an illustration.
-particle structs. This is because these particle variables are special and used
-internally by AMReX to assign the particles to grids and to mark particles as
-valid or invalid, respectively.
+The ability to store particle data in AoS form is provided for backward
```
In Sphinx, you could embed this into a box that pops out, e.g., `attention` or `important` or `note` or `hint`:
```rst
.. attention::

   The ability ...
```
https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-attention

ax3l commented on Docs/sphinx_documentation/source/Particle.rst:
```diff
@@ -438,6 +427,41 @@ on whether ``USE_SINGLE_PRECISION_PARTICLES`` is ``TRUE`` or not. We recommend
+the underlying :cpp:`StructOfArrays`, which can be used as before. However, it is
+particlularly convenient to use the :cpp:`[]` operator of :cpp:`ParticleTileData`, which
+allows the same code to work with both AoS and pure SoA particles. For example, within
+a `ParIter` loop, one can do:
```
rst is weird with verbatim (double quote needed):
```suggestion
a ``ParIter`` loop, one can do:
```
or
```suggestion
a :cpp:`ParIter` loop, one can do:
```

WeiqunZhang committed:
PODVector: Add extra capacity (#4734)

Add some extra room when we call `PODVector::resize` and `reserve`. The
extra capacity is computed as 3*sqrt(capacity) suggested by
@AlexanderSinn, and is capped at 10%. This might help particle codes
avoid memory re-allocation.

atmyers committed:
Remove old Fortran style from particle-mesh documentation (#4761)

The proposed changes:
- [ ] fix a bug or incorrect behavior in AMReX
- [ ] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX
users
- [x] include documentation in the code and/or rst files, if appropriate

atmyers committed:
Add information about generating particles for GPU codes to the docs (#4760)

The proposed changes:
- [ ] fix a bug or incorrect behavior in AMReX
- [ ] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX
users
- [x] include documentation in the code and/or rst files, if appropriate

---------

Co-authored-by: Axel Huebl <axel.huebl@plasma.ninja>

atmyers committed:
Update Docs/sphinx_documentation/source/Particle.rst

Co-authored-by: Axel Huebl <axel.huebl@plasma.ninja>

atmyers committed:
use attention box

atmyers committed:
Merge branch 'development' into doc_pure_soa
